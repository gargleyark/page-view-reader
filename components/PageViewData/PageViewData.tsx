import React from "react";
import useLogs from '../../context/LogsContext/useLogs';

interface PageViewDataProps {
  path: string;
  count: number;
  label: string;
}

const PageViewData = ({
  path, count, label
}: PageViewDataProps): JSX.Element => {
  return (
    <div>
      <span>{ path }</span>:{` `}
      <span>{ count } { label }</span>
    </div>
  );
}

export default PageViewData;