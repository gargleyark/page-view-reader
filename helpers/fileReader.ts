import { Log } from '../context/LogsContext/LogsContext';

export const formatLogs = (logs: string): Array<Log> => {
    return logs
        .split(/\n/)
        .filter(exists => !!exists)
        .map(log => ({
            page: log.split(' ')[0],
            ip: log.split(' ')[1],
        }))
}

export const getFileContent = (file: Blob) => {
    return new Promise<Array<Log>>((resolve) => {
        var fileReader = new FileReader();  
        fileReader.onloadend = () => resolve(formatLogs(`${fileReader.result}`));
        fileReader.onerror = () => resolve([]);
        fileReader.readAsText(file);
    });
};