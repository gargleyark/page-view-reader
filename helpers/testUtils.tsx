import { render } from "@testing-library/react";
import LogsContext, { Log } from '../context/LogsContext/LogsContext'

const Providers = ({ children }: { children: JSX.Element}) => {
  const logs: Array<Log> = [
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '2'
    },
    {
      page: '/about',
      ip: '2'
    }
  ];

  const setLogs = () => {};

  return (
    <LogsContext.Provider value={{ logs, setLogs }}>
      { children }
    </LogsContext.Provider >
  );
};

const customRender = (ui: JSX.Element, options: Record<string, string> = {}) =>
  render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";

export { customRender as render };