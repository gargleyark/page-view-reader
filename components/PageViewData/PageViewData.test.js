import React from "react";
import { render, screen } from "../../helpers/testUtils";
import PageViewData from ".";

describe("Pageview data", () => {
    it('should render correctly', () => {
        const { container } = render(<PageViewData path="/index" count={4} label="page views" />);
        expect(container).toMatchSnapshot();
    });
    it('should render text on page', () => {
        render(<PageViewData path="/index" count={4} label="page views" />);
        expect(screen.getByText('4 page views')).toBeInTheDocument();
        expect(screen.getByText('/index')).toBeInTheDocument();
    });
});