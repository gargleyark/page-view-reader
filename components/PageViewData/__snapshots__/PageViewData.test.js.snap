// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pageview data should render correctly 1`] = `
<div>
  <div>
    <span>
      /index
    </span>
    :
     
    <span>
      4
       
      page views
    </span>
  </div>
</div>
`;
